src/app/application.cpp

```diff
@@ -106,9 +123,105 @@ Application::Application(const QString &id, int &argc, char **argv)
     connect(this, SIGNAL(messageReceived(const QString &)), SLOT(processMessage(const QString &)));
     connect(this, SIGNAL(aboutToQuit()), SLOT(cleanup()));
 
+    if (isFileLoggerEnabled())
+        m_fileLogger = new FileLogger(getFileLoggerPath(), isFileLoggerBackup(), getFileLoggerMaxSize(), isFileLoggerDeleteOld(), getFileLoggerAge(), static_cast<FileLogger::FileLogAgeType>(getFileLoggerAgeType()));
+
     Logger::instance()->addMessage(tr("qBittorrent %1 started", "qBittorrent v3.2.0alpha started").arg(VERSION));
 }
 
+bool Application::isFileLoggerEnabled() const
+{
+    return settings()->loadValue(KEY_FILELOGGER_ENABLED, true).toBool();
+}
+
+void Application::setFileLoggerEnabled(bool value)
+{
+    if (value && !m_fileLogger)
+        m_fileLogger = new FileLogger(getFileLoggerPath(), isFileLoggerBackup(), getFileLoggerMaxSize(), isFileLoggerDeleteOld(), getFileLoggerAge(), static_cast<FileLogger::FileLogAgeType>(getFileLoggerAgeType()));
+    else
+        delete m_fileLogger;
+    settings()->storeValue(KEY_FILELOGGER_ENABLED, value);
+}
+
+QString Application::getFileLoggerPath() const
+{
+    return settings()->loadValue(KEY_FILELOGGER_PATH, QVariant(Utils::Fs::QDesktopServicesDataLocation() + LOG_FOLDER)).toString();
```

Comment:
Weird. What does `QStringBuilder<QString,QString>`? `operator+(const QString &, const QString &)` must return `const QString()` according to the documentation.